plugins/Jobs.cpp

```diff
@@ -739,38 +736,36 @@ bool BedrockPlugin_Jobs::processCommand(SQLite& db, BedrockCommand& command) {
                 retriableJobs.push_back(job);
             } else {
                 nonRetriableJobs.push_back(result[c][0]);
+            }
 
-                // Only non-retryable jobs can have children so see if this job has any
-                // FINISHED/CANCELLED child jobs, indicating it is being resumed
-                SQResult childJobs;
-                if (!db.read("SELECT jobID, data, state FROM jobs WHERE parentJobID != 0 AND parentJobID=" + result[c][0] + " AND state IN ('FINISHED', 'CANCELLED');", childJobs)) {
-                    STHROW("502 Failed to select finished child jobs");
-                }
+            // See if this job has any FINISHED/CANCELLED child jobs, indicating it is being resumed
+            SQResult childJobs;
+            if (!db.read("SELECT jobID, data, state FROM jobs WHERE parentJobID != 0 AND parentJobID=" + result[c][0] + " AND state IN ('FINISHED', 'CANCELLED');", childJobs)) {
+                STHROW("502 Failed to select finished child jobs");
+            }
 
-                if (!childJobs.empty()) {
-                    // Add associative arrays of all children depending on their states
-                    list<string> finishedChildJobArray;
-                    list<string> cancelledChildJobArray;
-                    for (auto row : childJobs.rows) {
-                        STable childJob;
-                        childJob["jobID"] = row[0];
-                        childJob["data"] = row[1];
-
-                        if (row[2] ==  "FINISHED") {
-                            finishedChildJobArray.push_back(SComposeJSONObject(childJob));
-                        } else {
-                            cancelledChildJobArray.push_back(SComposeJSONObject(childJob));
-                        }
+            if (!childJobs.empty()) {
+                // Add associative arrays of all children depending on their states
```

Comment:
I know it's the same old comment as before, but these are just plain old arrays, they're not associative, which is a weird term to use in C++ anyway.

Reply:
Fixed 😄